tests/management/group/test_view.py

```diff
@@ -2005,3 +2005,15 @@ def test_group_role_filter_as_non_admin(self):
         client = APIClient()
         response = client.get(url, **self.headers)
         self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
+
+    def test_remove_group_service_accounts_success(self):
+        """Test that removing the stub service accounts from a group returns successfully."""
+        test_user = Principal.objects.create(username="test_user", tenant=self.tenant)
+        self.group.principals.add(test_user)
+
+        url = reverse("group-principals", kwargs={"uuid": self.group.uuid})
+        client = APIClient()
+
+        url = "{}?service-accounts={}".format(url, "one,two,three")
+        response = client.delete(url, format="json", **self.headers)
+        self.assertEqual(response.status_code, status.HTTP_204_NO_CONTENT)
```

Comment:
This one fails because the endpoint returns a `403 Forbidden` status code.

Reply:
I have tried adding the `send_kafka_message` and `mock_request` arguments to the test, similar to what the following test had...

```python
    @patch(
        "management.principal.proxy.PrincipalProxy.request_filtered_principals",
        return_value={"status_code": 200, "data": [{"username": "test_user"}]},
    )
    @patch("core.kafka.RBACProducer.send_kafka_message")
    def test_remove_group_principals_success(self, send_kafka_message, mock_request):
```

But that doesn't make it work either :/

Reply:
Hey @MikelAlejoBR! It's because you've added these tests under the `GroupViewNonAdminTests`, which has different test setup from `GroupViewsetTests`. If you move the tests up into the `GroupViewsetTests` class, you shouldn't get 403s.

Reply:
Perfect, thank you for the suggestions! I was able to make it work :tada: